### ISSUE TITLE
Most vehicle data basically missing (counting drones and trains aswell)

### ISSUE THREAD

**Issue by tomatosauc** [porisius/FicsitRemoteMonitoring#99]
The getTrain endpoint of the api returns all trains as derailed, as the status variable, but derailed variable is false. Also train stations appear without any inventory (and were the train station status and train power consumption removed from the API?). Vehicles are also always airborne... Also drones have basically no data...

**Comment by cosmopetrich** [comment]
Train station inventories will currently only be shown if there is a docked carriage at the platform. There's some discussion about it in [this Discord thread](https://discord.com/channels/1142919853053841488/1186378513427730502).

I've spun up #101 to track that.

**Comment by V0IDL355** [comment]
> The getTrain endpoint of the api returns all trains as derailed, as the status variable, but derailed variable is false. Also train stations appear without any inventory (and were the train station status and train power consumption removed from the API?). Vehicles are also always airborne... Also drones have basically no data... 

Drones got some stuff commented out as from what I remember something got removed so Darth needs to find new ways to get that data if possible.

Seen the derailed issue personally, even if the train was moving.

**Comment by porisius** [comment]
What's the value of these three properties?

		JTrain->Values.Add("Derailed", MakeShared<FJsonValueBoolean>(Train->IsDerailed()));
		JTrain->Values.Add("PendingDerail", MakeShared<FJsonValueBoolean>(Train->HasPendingCollision()));
		JTrain->Values.Add("Status", MakeShared<FJsonValueString>(FormString));

**Comment by tomatosauc** [comment]
These are the properties if I am not mistaken... The train's running just fine, but the status is not right
```
"Derailed": false,
"PendingDerail": false,
"Status": "Derailed"
```

**Comment by porisius** [comment]
That's looking more like a game bug to me.

I am going to conduct some tests

**Comment by porisius** [comment]
Freshly built train... never moved... game bug

    "Derailed": false,
    "PendingDerail": false,
    "Status": "Derailed",

**Comment by porisius** [comment]
Haha, apparently not... I made a log, and apparently I need to fix something.
[2024.10.23-18.26.50:984][  4]LogFRMAPI: Warning: Train Status ENUM: ETrainStatus::TS_SelfDriving

**Comment by featheredtoast** [comment]
Just glancing quickly at the code, C++ should be using `==` for comparison. `=` is setting the value. This looks like it affects all similar switch statements, not just trains.

https://github.com/porisius/FicsitRemoteMonitoring/blob/47269f391d45337da3ee4b27c69e469c1e31ba71/Source/FicsitRemoteMonitoring/Private/FRM_Trains.cpp#L96-L99

**Comment by porisius** [comment]
Its a switch/case, and based on that, it will set the FString variable to the text. I found and got a fix.

**Comment by featheredtoast** [comment]
Realized that the moment I posted it. 🤦  carry on

**Comment by porisius** [comment]
Fixed in 1.1.0
